### PR TITLE
Add FXIOS-5733 [v112] Alerts in edit card view

### PR DESCRIPTION
**Pull request by nishant2718**
# [FXIOS-5721](https://mozilla-hub.atlassian.net/browse/FXIOS-5721)
## Overview
TLDR: Add the alert view for the CC Edit View.

## Details
We forego the approach of creating a custom alert view, given that Alerts are changing quite a bit between versions. It seems  the native solution is flexible enough for our needs in the future. 

We use Alert here, but it'll need updates to follow along with the updates SwiftUI's alerts have undergone in later versions.

Button actions are left as empty closures for now. In a future ticket, when UX is clean and the datasource integration begins, those closures will have the right methods to remove cards locally and in AS. 

We `init` inject `Profile` because a preview is using that one. In normal cases, I expect `Profile` to be passed along like the rest of the app is doing.

### CODE REVIEW

File: Client/Frontend/Autofill/CreditCard/CreditCardSettingsView/CreditCardEditViewModel.swift [hunk 1]
```diff
@@ -31,12 +35,39 @@ class CreditCardEditViewModel: ObservableObject {
...
+            secondaryButtonAction: {}
+        )
+    }
+
+    init(profile: Profile) {
+        self.profile = profile
+    }
 
-    init(firstName: String,
+    init(profile: Profile = AppContainer.shared.resolve(),
```
Comment (nishant2718) [inline Client/Frontend/Autofill/CreditCard/CreditCardSettingsView/CreditCardEditViewModel.swift:65]: This init's used in the previews only so far. So I figured init inject of profile would be harmless here.

**Comment by mte-test** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Messages</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:book:</td>
      <td>Project coverage: 29.23%</td>
    </tr>
  
<tr>
      <td>:book:</td>
      <td>Edited 5 files</td>
    </tr>
  
<tr>
      <td>:book:</td>
      <td>Created 0 files</td>
    </tr>
  </tbody>
</table>

## Client.app: Coverage: 25.83
| File | Coverage ||
| --- | --- | --- |
CreditCardEditView.swift | 0.0% | ⚠️
CreditCardEditViewModel.swift | 0.0% | ⚠️
CreditCardSettingsViewModel.swift | 0.0% | ⚠️
RemoveCardButton.swift | 0.0% | ⚠️

<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/swift">Danger Swift</a> against f0f03dde3769db092c252ee471f19f937e649ebb
</p>

**Review by thatswinnie — APPROVED** [review]

**Comment by nishant2718** [comment]
image

The difference b/w that BR run and the next commit is just spacing. To save credits, I won't run BR for the spacing change from the last commit.

**Review by nbhasin2 — APPROVED** [review]
